packages/ui/src/views/repo/pull-request/compare/pull-request-compare-page.tsx

```diff
@@ -214,34 +210,8 @@ export const PullRequestComparePage: FC<PullRequestComparePageProps> = ({
           </p>
           <Layout.Horizontal className="items-center" gap="gap-x-2.5">
             <Icon name="compare" size={14} className="text-icons-1" />
-            <BranchSelector
-              isBranchOnly={true}
-              useTranslationStore={useTranslationStore}
-              useRepoBranchesStore={useRepoBranchesStore}
-              branchPrefix="base"
-              selectedBranch={targetBranch}
-              onSelectBranch={(branchTag, type) => {
-                selectBranch(branchTag, type, false)
-                handleBranchSelection()
-              }}
-              searchQuery={searchTargetQuery}
-              setSearchQuery={setSearchTargetQuery}
-            />
 
-            <Icon name="arrow-long" size={12} className="rotate-180 text-icons-1" />
-            <BranchSelector
-              isBranchOnly={true}
-              useTranslationStore={useTranslationStore}
-              useRepoBranchesStore={useRepoBranchesStore}
-              branchPrefix="compare"
-              selectedBranch={sourceBranch}
-              onSelectBranch={(branchTag, type) => {
-                selectBranch(branchTag, type, true)
-                handleBranchSelection()
-              }}
-              searchQuery={searchSourceQuery}
-              setSearchQuery={setSearchSourceQuery}
-            />
+            {branchSelector()}
```

Comment:
variables that start with capital letters can be rendered as components directly
```suggestion
            <BranchSelector />
```

Reply:
I don't think this is possible since we're directly passing down a `ReactElement`?

in App:
```
branchSelectorRenderer = {
    <BranchSelector props={props} />
}
```

in Views:
```
{branchSelectorRenderer}
```